lib/dialects/abstract/query-generator.js

```diff
@@ -179,6 +178,20 @@ class QueryGenerator {
       }
     }
 
+    let onDuplicateKeyUpdate = '';
+
+    if (this._dialect.supports.inserts.updateOnDuplicate && options.updateOnDuplicate) {
+      if (this._dialect.supports.inserts.updateOnDuplicate == ' ON CONFLICT DO UPDATE SET') { // postgres / sqlite
+        // If no conflict target columns were specified, use the primary key names from options.upsertKeys
+        const conflictKeys = options.upsertKeys.map(attr => this.quoteIdentifier(attr));
+        const updateKeys = options.updateOnDuplicate.map(attr => `${this.quoteIdentifier(attr)}=EXCLUDED.${this.quoteIdentifier(attr)}`);
```

Comment:
Nit: Result of `quoteIdentifier` could be cached to avoid 2 calls.

Reply:
This is actually a c/p from `bulkInsert` above, I have plan to fix both of them. So I'll like to keep it just as original for now.